hack/e2e/setup/setup_test.go

```diff
@@ -0,0 +1,483 @@
+	"github.com/kyma-project/eventing-manager/hack/e2e/env"
+	"github.com/kyma-project/eventing-manager/pkg/eventing"
+	"github.com/pkg/errors"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/kyma-project/eventing-manager/hack/e2e/common"
+	. "github.com/kyma-project/eventing-manager/hack/e2e/common/fixtures"
+)
+
+// Constants for retries.
+const (
+	interval = 2 * time.Second
+	attempts = 60
+)
+
+// clientSet is what is used to access K8s build-in resources like Pods, Namespaces and so on.
+var clientSet *kubernetes.Clientset //nolint:gochecknoglobals // This will only be accessible in e2e tests.
+
+// k8sClient is what is used to access the Eventing CR.
+var k8sClient client.Client //nolint:gochecknoglobals // This will only be accessible in e2e tests.
+
+var logger *zap.Logger
+
+var testConfigs env.E2EConfig
+
+// TestMain runs before all the other test functions. It sets up all the resources that are shared between the different
+// test functions. It will then run the tests and finally shuts everything down.
+func TestMain(m *testing.M) {
+	var err error
+	logger, err = SetupLogger()
+	if err != nil {
+		logger.Error(err.Error())
+		panic(err)
+	}
+
+	testConfigs, err = env.GetE2EConfig()
+	if err != nil {
+		logger.Error(err.Error())
+		panic(err)
+
+	}
+	logger.Info(fmt.Sprintf("##### NOTE: Tests will run w.r.t. backend: %s", testConfigs.BackendType))
+
+	clientSet, k8sClient, err = GetK8sClients()
+	if err != nil {
+		logger.Error(err.Error())
+		panic(err)
+	}
+
+	ctx := context.TODO()
+	// Create the Namespace used for testing.
+	err = Retry(attempts, interval, func() error {
+		// It's fine if the Namespace already exists.
+		return client.IgnoreAlreadyExists(k8sClient.Create(ctx, Namespace()))
+	})
+	if err != nil {
+		logger.Error(err.Error())
+		panic(err)
+	}
+
+	// Wait for eventing-manager deployment to get ready.
+	if testConfigs.ManagerImage == "" {
+		logger.Warn(
+			"ENV `MANAGER_IMAGE` is not set. Test will not verify if the " +
+				"manager deployment image is correct or not.",
+		)
+	}
+	if err := waitForEventingManagerDeploymentReady(testConfigs.ManagerImage); err != nil {
+		logger.Error(err.Error())
+		panic(err)
+	}
+
+	// Create the Eventing CR used for testing.
+	err = Retry(attempts, interval, func() error {
+		eventingCR := EventingCR(eventingv1alpha1.BackendType(testConfigs.BackendType))
+		errEvnt := k8sClient.Create(ctx, eventingCR)
+		if k8serrors.IsAlreadyExists(errEvnt) {
+			gotEventingCR, getErr := getEventingCR(ctx, eventingCR.Name, eventingCR.Namespace)
+			if getErr != nil {
+				return err
+			}
+
+			// If Backend type is changed then update the CR.
+			if gotEventingCR.Spec.Backend.Type != eventingCR.Spec.Backend.Type {
```

Comment:
Do we expect the backend type to change? Why? What happens if it changed after this code is already executed?

Reply:
It is only the the switching tests. We will change the CR manually and run the `e2e-setup` target again. At that time it will update the CR.